code/modules/antagonists/eldritch_cult/eldritch_items.dm

```diff
@@ -16,20 +16,51 @@
 		return
 	var/dist = get_dist(user.loc,target.loc)
 	var/dir = get_dir(user.loc,target.loc)
-
-	switch(dist)
-		if(0 to 15)
-			to_chat(user,"<span class='warning'>[target.real_name] is near you. They are to the [dir2text(dir)] of you!</span>")
-		if(16 to 31)
-			to_chat(user,"<span class='warning'>[target.real_name] is somewhere in your vicinty. They are to the [dir2text(dir)] of you!</span>")
-		if(32 to 127)
-			to_chat(user,"<span class='warning'>[target.real_name] is far away from you. They are to the [dir2text(dir)] of you!</span>")
-		else
-			to_chat(user,"<span class='warning'>[target.real_name] is beyond our reach.</span>")
+	if(user.z != target.z)
+		to_chat(user,"<span class='warning'>[target.real_name] is on another plane of existance!</span>")
+	else
+		switch(dist)
+			if(0 to 15)
+				to_chat(user,"<span class='warning'>[target.real_name] is near you. They are to the [dir2text(dir)] of you!</span>")
+			if(16 to 31)
+				to_chat(user,"<span class='warning'>[target.real_name] is somewhere in your vicinty. They are to the [dir2text(dir)] of you!</span>")
+			if(32 to 127)
+				to_chat(user,"<span class='warning'>[target.real_name] is far away from you. They are to the [dir2text(dir)] of you!</span>")
+			else
+				to_chat(user,"<span class='warning'>[target.real_name] is beyond our reach.</span>")
 
 	if(target.stat == DEAD)
 		to_chat(user,"<span class='warning'>[target.real_name] is dead. Bring them onto a transmutation rune!</span>")
 
+/datum/action/innate/heretic_shatter
```

Comment:
this kinda feels like something that should be a "z"/use in-hand ability, not (or at least in addition to being) its own action button

Reply:
i feel like it could be missclicked. This action literally destroys your weapon and teles you to a random part of the z level, the choice to do so must be deliberate.

Reply:
fair enough.